src/Microsoft.Azure.ServiceBus/Primitives/ServiceBusConnection.cs

```diff
@@ -100,5 +93,49 @@ static void CloseConnection(AmqpConnection connection)
 
             return connection;
         }
+
+        private TransportSettings CreateTransportSettings()
+        {
+            string hostName = this.Endpoint.Host;
+            string networkHost = this.Endpoint.Host;
+            int port = this.Endpoint.Port;
+            string scheme = this.Endpoint.Scheme;
+
+            if (scheme.Equals(WebSocketConstants.WebSocketSecureScheme, StringComparison.OrdinalIgnoreCase))
+            {
+                return AmqpConnectionHelper.CreateWebSocketTransportSettings(
+                    networkHost: networkHost,
+                    hostName: hostName,
+                    port: port);
+            }
+            else
+            {
+                return AmqpConnectionHelper.CreateTcpTransportSettings(
+                    networkHost: networkHost,
+                    hostName: hostName,
+                    port: port,
+                    useSslStreamSecurity: true);
+            }
+        }
+
+        private static Task<TransportBase> ConnectTaskAsync(TransportInitiator transportInitiator, TimeoutHelper timeoutHelper)
```

Comment:
>ConnectTaskAsync [](start = 43, length = 16)

This is pretty much what `AmqpTransportInitiator` does. I do not see a reason why we need to re-write this in our code. Its a lot of code duplication. Could you revert this method and get back to 
`AmqpTransportInitiator initiator = new AmqpTransportInitiator(amqpSettings, tpSettings);`
 `TransportBase transport = await initiator.ConnectTaskAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);`